### PR TITLE
Add Cloud Deploy Delivery Pipeline

### PR DESCRIPTION
**Pull request by nownabe**
Part of https://github.com/hashicorp/terraform-provider-google/issues/13042

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_clouddeploy_delivery_pipeline_iam_policy`
`google_clouddeploy_delivery_pipeline_iam_binding`
`google_clouddeploy_delivery_pipeline_iam_member`
```

### CODE REVIEW

File: mmv1/products/clouddeploy/DeliveryPipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            name: "status"
+            description: "True if the targets are all a comparable type. For example this is true if all targets are GKE clusters. This is false if some targets are Cloud Run targets and others are GKE clusters."
+            output: true
+          - !ruby/object:Api::Type::String
+            name: "errorDetails"
+            description: "Human readable error message."
+            output: true
+  - !ruby/object:Api::Type::String
+    name: "etag"
+    description: "This checksum is computed by the server based on the value of other fields, and may be sent on update and delete requests to ensure the client has an up-to-date value before proceeding."
```
Comment (c2thorn) [inline mmv1/products/clouddeploy/DeliveryPipeline.yaml:160]: If this field will be provided by the API but still an input field, we should add `default_from_api: true`

File: mmv1/products/clouddeploy/DeliveryPipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  - !ruby/object:Api::Type::Boolean
+    name: "suspended"
```
Comment (c2thorn) [inline mmv1/products/clouddeploy/DeliveryPipeline.yaml:162]: Is this some sort of state field that will be returned by the API? Or purely input?

File: mmv1/templates/terraform/examples/clouddeploy_delivery_pipeline_basic.tf.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+resource "google_clouddeploy_delivery_pipeline" "<%= ctx[:primary_resource_id] %>" {
+  name = "<%= ctx[:vars]['delivery_pipeline'] %>"
+  location = "us-central1"
+  serial_pipeline  {
+    stages {
+      target_id = "test"
+      profiles = []
```
Comment (c2thorn) [inline mmv1/templates/terraform/examples/clouddeploy_delivery_pipeline_basic.tf.erb:7]: where did this `profile` fields come from? There does not seem to be a corresponding field in the schema.

**Review by c2thorn — CHANGES_REQUESTED** [review]
About to run the CI+tests, but already have some requests. Can you add a[ handwritten test](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-a-create-test) that updates all updatable fields?

Also, it looks like the current example will fail, see my comment inline.

File: mmv1/products/clouddeploy/DeliveryPipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,178 @@
+# Copyright 2023 Google Inc.
```
Comment (c2thorn) [inline mmv1/products/clouddeploy/DeliveryPipeline.yaml:1]:
```suggestion
# Copyright 2024 Google Inc.
```

**Review by c2thorn — CHANGES_REQUESTED** [review]
I see now that this is likely added just to add the IAM resources, not the actual Cloud Deploy Delivery Pipeline resource, which is already implemented as a DCL resource.
If this is the case, then we should add `exclude_resource: true` to stop all resource generation and just have IAM generation run. You can refer to this file as an example: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/cloudbuildv2/Connection.yaml#L18

**Comment by nownabe** [comment]
@c2thorn Thank you for your review! I'll check your comment and try to fix

File: mmv1/products/clouddeploy/DeliveryPipeline.yaml (hunk 2 above)
Comment (nownabe) [reply mmv1/products/clouddeploy/DeliveryPipeline.yaml:162]: `suspended` is an input field.
https://cloud.google.com/deploy/docs/suspend-pipeline

File: mmv1/templates/terraform/examples/clouddeploy_delivery_pipeline_basic.tf.erb (hunk 3 above)
Comment (nownabe) [reply mmv1/templates/terraform/examples/clouddeploy_delivery_pipeline_basic.tf.erb:7]: oops. I should add [correct stages](https://cloud.google.com/deploy/docs/api/reference/rest/v1/projects.locations.deliveryPipelines#DeliveryPipeline.SerialPipeline) into `DeliveryPipeline.yaml`

**Comment by nownabe** [comment]
@c2thorn I added `exclude_resource: true` and deleted properties. Should I still add handwritten tests?

**Comment by c2thorn** [comment]
> @c2thorn I added `exclude_resource: true` and deleted properties. Should I still add handwritten tests?

No, that should be enough!

File: mmv1/templates/terraform/examples/clouddeploy_delivery_pipeline_basic.tf.erb (hunk 3 above)
Comment (c2thorn) [reply mmv1/templates/terraform/examples/clouddeploy_delivery_pipeline_basic.tf.erb:7]: this example should not be needed at all. You can delete the file and `examples` block in the YAML

**Comment by nownabe** [comment]
@c2thorn Is there anything else I need to do?

**Review by c2thorn — APPROVED** [review]

**Pull request merged**